### PR TITLE
Decompile (non-matching) sub_8002B20.

### PR DESCRIPTION
**Pull request by JaceCear**
* Document a few things in Background struct

### CODE REVIEW

File: include/sprite.h [hunk 1]
```diff
@@ -56,10 +76,12 @@ typedef struct {
     // Flags
     // 0x200 = something about updating animations (sub_8003638)
     u16 unk2E;
-    u16 unk30;
-    u16 unk32;
-    u16 unk34;
-    u16 unk36;
+
+    // apparently NOT signed?
```
Comment (freshollie) [inline include/sprite.h:80]: Maybe we don't have any instances which prove they are not signed?

File: include/sprite.h [hunk 2]
```diff
@@ -95,16 +117,18 @@ typedef struct {
...
 #define SPRITE_MAYBE_SWITCH_ANIM(sprite)                                                \
-    if (((sprite)->unk21 != (sprite)->variant)                                          \
-        || ((sprite)->unk1E != (sprite)->graphics.anim)) {                              \
-        (sprite)->graphics.size = 0;                                                    \
-        (sprite)->unk21 = (sprite)->variant;                                            \
-        (sprite)->unk1E = (sprite)->graphics.anim;                                      \
-        (sprite)->unk14 = 0;                                                            \
-        (sprite)->unk1C = 0;                                                            \
-        (sprite)->unk10 &= ~0x4000;                                                     \
+    if ( SpriteShouldUpdate(sprite) {                   \
```
Comment (freshollie) [inline include/sprite.h:125]: Isn't there a bracket missing here? I'm a bit confused where the closing bracket of the `if` is?

File: src/bg_update.c [hunk 3]
```diff
@@ -0,0 +1,561 @@
+#include "global.h"
+#include "gba/syscall.h"
```
Comment (freshollie) [inline src/bg_update.c:2]: I think this is included as part of globals anyway?

File: src/bg_update.c [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+                            }
+                        } else {
+                            // _08003254
+                            u32 index = sp14;
+                            u16 *r0Ptr = (u16 *)&((u8 *)bg->unk10)[r4 * index * sp08];
+                            u32 index2 = sp10;
+                            u16 *r4Ptr = &r0Ptr[index2 * sp08];
+
+                            // _08003298
+                            while (--r5 != (u16)-1) {
```
Comment (freshollie) [inline src/bg_update.c:354]: Are these `CopyLarge`?

File: ldscript.txt [hunk 5]
```diff
@@ -34,7 +34,7 @@ SECTIONS
         build/sa2/src/main.o(.text);
         build/sa2/src/task.o(.text);
         build/sa2/src/sprite_1.o(.text);
-        build/sa2/asm/sprite.o(.text);
+        build/sa2/src/bg_update.o(.text);
```
Comment (freshollie) [inline ldscript.txt:37]: I think maybe we'll end up following katam here and naming `bg.c`, it's fine for now

File: include/sprite.h [hunk 6]
```diff
@@ -95,16 +117,18 @@ typedef struct {
 
 #define SPRITE_BF_GET_BG_ID(sprite) (((sprite)->unk10 & 0x18000) >> 15)
 
-// TODO: Maybe rename this?
+#define SpriteShouldUpdate(sprite) ((sprite)->unk21 != (sprite)->variant) || \
```
Comment (freshollie) [inline include/sprite.h:120]: I think there is a missing bracket here making this weird?

**Review by freshollie — COMMENTED** [review]
Great work, only small confusion with the macro which I think needs fixing (though I understand it compiles just hard to read)

File: include/sprite.h (hunk 2 above)
Comment (JaceCear) [reply include/sprite.h:125]: That's correct! Weird that it compiled just fine...

File: include/sprite.h (hunk 1 above)
Comment (JaceCear) [reply include/sprite.h:80]: I don't know. I'll leave the comment there for now

File: src/bg_update.c (hunk 4 above)
Comment (JaceCear) [reply src/bg_update.c:354]: I don't see a macro that's called that

**Comment by JaceCear** [comment]
Is this outdated or not?
https://github.com/freshollie/sa2/blob/sprite_stuff/include/sprite.h#L35-L36

File: src/bg_update.c (hunk 4 above)
Comment (freshollie) [reply src/bg_update.c:353]: https://github.com/freshollie/sa2/blob/main/include/gba/macro.h#L87

it's probably not this, but maybe in other cases where we are doing `DmaCopy` in loops

**Comment by freshollie** [comment]
Yes, out of date

**Review by freshollie — APPROVED** [review]

**Pull request merged**